**Ejercicio 2.** Considera la curva $\mathbf{z}(t)=(t,t^2,t^3)$ con $t\in[-1,1]$.

1.  Calcula el triedro de Frenet-Serret, la curvatura y la torsión.
2.  Dibuja la curva $\mathbf{z}$ en $[-1,1]$ con el triedro en una animación.
3.  Calcula (numéricamente si es preciso) la longitud de la curva.
4.  Toma tres valores del parámetro $t=t_1,t_2,t_3$. Comprueba que la intersección de los planos osculadores para esos valores del parámetro está en el plano que contiene a $\mathbf{z}(t_i)$, $i=1,2,3$. 
5. Elige tres valores aleatorios y dibuja los planos del apartado anterior.
6. Encuentra valores $a,b,c\in\mathbb{R}_{>0}$ tales que la función longitud de $\mathbf{z}(t)=(at,bt^2,ct^3)$ se puede calcular de manera elemental.


$\textbf{Apartado 1}$

In [11]:
reset ()
var('x,y,z,t,s,a,b,c,t1,t2,t3',domain='real')
V=vector([t,t^2,t^3])

In [12]:
def curvaespacial(V,vr=t):
    V1,V2,V3=[V.derivative(vr,j) for j in (1,2,3)]
    V12=V1.cross_product(V2)
    reg=(V1*V1).simplify_full()
    ct=V1.normalized().canonicalize_radical().simplify_full()
    cn=ct.derivative(vr).normalized().canonicalize_radical().simplify_full()
    cb=V12.normalized().canonicalize_radical().simplify_full()
    kappa=V12.norm()/V1.norm()^3
    kappa=kappa.canonicalize_radical().simplify_full()
    tau=Matrix([V1,V2,V3]).det()/(V12*V12)
    tau=tau.simplify_full()
    return V1,V2,V3,V12,reg,ct,cn,cb,kappa,tau

In [13]:
V1,V2,V3,V12,reg,ct,cn,cb,kappa,tau=curvaespacial(V)

show(ct,cn,cb)
show(kappa)
show(tau)

(1/sqrt(9*t^4 + 4*t^2 + 1), 2*t/sqrt(9*t^4 + 4*t^2 + 1), 3*t^2/sqrt(9*t^4 + 4*t^2 + 1)) (-(9*t^3 + 2*t)/(sqrt(9*t^4 + 9*t^2 + 1)*sqrt(9*t^4 + 4*t^2 + 1)), -(9*t^4 - 1)/(sqrt(9*t^4 + 9*t^2 + 1)*sqrt(9*t^4 + 4*t^2 + 1)), 3*(2*t^3 + t)/(sqrt(9*t^4 + 9*t^2 + 1)*sqrt(9*t^4 + 4*t^2 + 1))) (3*t^2/sqrt(9*t^4 + 9*t^2 + 1), -3*t/sqrt(9*t^4 + 9*t^2 + 1), 1/sqrt(9*t^4 + 9*t^2 + 1))

2*sqrt(9*t^4 + 9*t^2 + 1)/(9*t^4 + 4*t^2 + 1)^(3/2)

3/(9*t^4 + 9*t^2 + 1)

$\to$ Obtenemos, en primera instancia, los vectores tangencial, normal y binormal, que conforman el triedro.
Luego, la curvatura y la torsión.

$\textbf{Apartado 2}$

Dibujamos la curva en el intervalo que se nos pide, $[-1,1]$.

In [14]:
P=parametric_plot3d(V,(t,-1,1), frame=False)
def flecha(V,cc,t0,color='red'):
    return arrow3d(V(t=t0),V(t=t0)+5*cc(t=t0),color=color)
A=animate([P+flecha(V,ct,t0)+flecha(V,cn,t0,color='purple')+flecha(V,cb,t0,color='green') for t0 in [-1,-0.9,..,1]])

In [15]:
A.interactive()

Graphics3d Object

$\textbf{Apartado 3}$

Calcularemos la longitud numéricamente.

In [16]:
c=V1.norm().simplify_full()
show(c)

sqrt(9*t^4 + 4*t^2 + 1)

In [17]:
d=(numerical_integral(c,-1,1))
show(d)

(3.726045965024503, 4.136742021181864e-14)

$\to$ Obtenemos el valor de la longitud: $3.726...$

$\textbf{Apartado 4}$

In [18]:
show(V12/2)

(3*t^2, -3*t, 1)

In [19]:
show((V12/2)*V)

t^3

$V12$ es el producto vectorial de la derivada segunda y la primera, así que es paralelo al binormal. Lo tomamos porque no nos importa que esté normalizado.
Los puntos el el plano tangente en $z(t)$ son los que cumplen la siguiente ecuación: $V12*((x,y,z)-z(t))=0$, es decir los perpendiculares al binormal.

La ecuación del plano es $\mathbf{V12*(x,y,z)=t^3}$
Aplicamos está ecuación con t1,t2 y t3 y resolvemos un sistema para obtener la intersección de los tres planos:
$$3{t_1}^2  x - 3{t_1}  y + z = {t_1}^3$$
$$3{t_2}^2  x - 3{t_2}  y + z = {t_2}^3$$
$$3{t_3}^2  x - 3{t_3}  y + z = {t_3}^3$$


In [20]:
sols=solve ([3*t1^2*x -3*t1*y+ 1*z==t1^3,3*t2^2*x -3*t2*y+ 1*z==t2^3, 3*t3^2*x -3*t3*y+ 1*z==t3^3],x,y,z)
show(sols)

[[x == 1/3*t1 + 1/3*t2 + 1/3*t3, y == 1/3*t1*(t2 + t3) + 1/3*t2*t3, z == t1*t2*t3]]

Resolvemos otro sistema para que el punto de intersección pertenezca al plano que contiene a $z(t1),z(t2),z(t3)$.$(V(t)==z(t))$
$$ V(t_1)+a(V(t_2)-V(t_1))+b*(V(t_3)-V(t_1))==sols$$

$$t_1+a(t_2-t_1)+b(t_3-t_1)=t_1/3+t_2/3+t_3/3$$
$$t_1^2+a(t_2^2-t_1^2)+b(t_3^2-t_1^2)=(t_1/3)*(t_2+t_3)+t_2*t_3/3$$
$$t_1^3+a(t_2^3-t_1^3)+b(t_3^3-t_1^3)=t_1t_2t_3$$

In [21]:
m=solve ([t1+a*(t2-t1)+b*(t3-t1)==t1/3+t2/3+t3/3, t1^2+a*(t2^2-t1^2)+b*(t3^2-t1^2)==(t1/3)*(t2+t3)+t2*t3/3,t1^3+a*(t2^3-t1^3)+b*(t3^3-t1^3)==t1*t2*t3],a,b)
show(m)

[[a == 1/3*(t1^2 - 2*t1*t3 + t3^2)/(t1*t2 - t2^2 - (t1 - t2)*t3), b == -1/3*(t1^2 - 2*t1*t2 + t2^2)/(t1*t2 - (t1 + t2)*t3 + t3^2)]]

El sistema tiene solución, así que podemos deducir que el punto de la intersección pertenece al plano.

$\textbf{Apartado 5}$

In [23]:
t1=-1
t2=0
t3=1

Q=V+a*ct+b*cn
P=vector([t1+a*(t2-t1)+b*(t3-t1), t1^2+a*(t2^2-t1^2)+b*(t3^2-t1^2),t1^3+a*(t2^3-t1^3)+b*(t3^3-t1^3)])

Q1=parametric_plot3d(Q(t=-1),(a,-4,4),(b,-4,4),color='red')
Q2=parametric_plot3d(Q(t=0),(a,-4,4),(b,-4,4),color='blue')
Q3=parametric_plot3d(Q(t=1),(a,-4,4),(b,-4,4),color='green')
P1=parametric_plot3d(P,(a,-5,5),(b,-5,5),color='purple')


Q1+Q2+Q3+P1

Graphics3d Object

$\textbf{Apartado 6}$

In [24]:
z=vector([a*t,b*t^2,c*t^3])
z1=z.derivative(t) 
show(z1)
zn=z1.norm()
show(zn)

(a, 2*b*t, 2*(9*t^3 + 2*t)*t^3/sqrt(9*t^4 + 4*t^2 + 1) + 3*sqrt(9*t^4 + 4*t^2 + 1)*t^2)

sqrt(4*b^2*t^2 + (2*(9*t^3 + 2*t)*t^3/sqrt(9*t^4 + 4*t^2 + 1) + 3*sqrt(9*t^4 + 4*t^2 + 1)*t^2)^2 + a^2)

Para calcular la integral de manera elemental necesitamos que la expresión sea un cuadrado perfecto con objeto de cancelar la raíz.

Quedaría el módulo de un polinomio de grado $2$, podríamos ver donde es positivo y negativo y separar la integral.

Debido a que aparece un coeficiente a la cuarta deducimos que será $(3ct^2+a)^2$.

Se deberá cumplir $4b^2=2^3ac$

$b=\sqrt{(3/2)ac}$

Un ejemplo sería $a=3/2 b=3/2 c=1$